#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_check_disk_smb_spaces.dpatch by
## Matthias Eble <psychotrahe@users.sourceforge.net>
##
## From 0892c4ac28c4c2e3b6275187f845813b4cd6d320 Mon Sep 17 00:00:00 2001
## From: Matthias Eble <psychotrahe@users.sourceforge.net>
## Date: Sun, 28 Nov 2010 21:35:59 +0100
## Subject: [PATCH] Make check_disk_smb accept spaces in share names
## (#990948, #1370031, Debian #601699)
##
## DP: Enables Support sharenames with spaces (http://bugs.debian.org/601699)
## DP: Upstream bug is: http://sourceforge.net/tracker/?func=detail&aid=990948&group_id=29880&atid=397597

@DPATCH@

diff --git a/plugins-scripts/check_disk_smb.pl b/plugins-scripts/check_disk_smb.pl
index 7c81fc2..4698700 100755
--- a/plugins-scripts/check_disk_smb.pl
+++ b/plugins-scripts/check_disk_smb.pl
@@ -67,7 +67,7 @@ my $host = $1 if ($opt_H =~ /^([-_.A-Za-z0-9 ]+\$?)$/);
 ($host) || usage("Invalid host: $opt_H\n");
 
 ($opt_s) || ($opt_s = shift @ARGV) || usage("Share volume not specified\n");
-my $share = $1 if ($opt_s =~ /^([-_.A-Za-z0-9]+\$?)$/);
+my $share = $1 if ($opt_s =~ /^([-_.A-Za-z0-9 ]+\$?)$/);
 ($share) || usage("Invalid share: $opt_s\n");
 
 defined($opt_u) || ($opt_u = shift @ARGV) || ($opt_u = "guest");

